logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/e2ei/usecase/GetMembersE2EICertificateStatusesUseCase.kt

```diff
@@ -36,17 +39,18 @@ interface GetMembersE2EICertificateStatusesUseCase {
 
 class GetMembersE2EICertificateStatusesUseCaseImpl internal constructor(
     private val mlsConversationRepository: MLSConversationRepository,
-    private val certificateStatusMapper: CertificateStatusMapper
+    private val certificateStatusMapper: CertificateStatusMapper,
+    private val clientRepository: ClientRepository
 ) : GetMembersE2EICertificateStatusesUseCase {
     override suspend operator fun invoke(conversationId: ConversationId, userIds: List<UserId>): Map<UserId, CertificateStatus?> =
-        mlsConversationRepository.getMembersIdentities(conversationId, userIds).fold(
-            { mapOf() },
-            {
-                it.mapValues { (_, identities) ->
-                    identities.getUserCertificateStatus(certificateStatusMapper)
+        mlsConversationRepository.getMembersIdentities(conversationId, userIds).flatMap { identitiesMap ->
+            clientRepository.getClientsByConversationId(conversationId).map { clientsMap ->
```

Comment:
Question: there is a problem here, we don't have the list of clients of the user since sometimes we might get out of sync with the cloud and the list of the clients, there are two solutions, 
one is we make sure we have always the updated list of the users clients
other one: since CC always has the list of the mls clients in the mls tree, we can ask the CC to include an aggregated user status based on all of their existence clients in the mls tree, or return all the clients even some of them doesn't have a wire-identity.

Reply:
yeah, i like the second too, but it may take some time and iOS and Web already added the fix on their sides.